internal/commands/app/create.go

```diff
@@ -354,11 +342,102 @@ func (cmd *CommandCreate) Handler(profile *user.Profile, ui terminal.UI, clients
 	return nil
 }
 
+func (cmd CommandCreate) handleCreateTemplateApp(
+	profile *user.Profile,
+	ui terminal.UI,
+	clients cli.Clients,
+	groupID string,
+	rootDir string,
+	dsClusters []dataSourceCluster,
+	dsDatalakes []dataSourceDatalake,
+) error {
+	if cmd.inputs.DryRun {
+		logs := make([]terminal.Log, 0, len(dsClusters)+len(dsDatalakes)+2)
+		logs = append(logs, terminal.NewTextLog(fmt.Sprintf(
+			"A Realm app would be created at %s using the '%s' template",
+			rootDir,
+			cmd.inputs.Template,
+		)))
+
+		for _, cluster := range dsClusters {
+			logs = append(logs, terminal.NewTextLog("The cluster '%s' would be linked as data source '%s'", cluster.Config.ClusterName, cluster.Name))
+		}
+		for _, datalake := range dsDatalakes {
+			logs = append(logs, terminal.NewTextLog("The data lake '%s' would be linked as data source '%s'", datalake.Config.DatalakeName, datalake.Name))
+		}
+		logs = append(logs, terminal.NewFollowupLog("To create this app run", cmd.display(true)))
+		ui.Print(logs...)
+		return nil
+	}
+
+	if len(dsClusters) != 1 {
+		return errors.New("could not send app creation request due to unexpected number of clusters")
+	}
+	createAppMetadata := realm.AppMeta{
+		Location:        cmd.inputs.Location,
+		DeploymentModel: cmd.inputs.DeploymentModel,
+		Environment:     cmd.inputs.Environment,
+		Template:        cmd.inputs.Template,
+		DataSource:      dsClusters[0],
+	}
+
+	createTemplateApp := func() (realm.App, error) {
+		s := spinner.New(terminal.SpinnerCircles, 250*time.Millisecond)
+		s.Suffix = " Creating template app..."
+		s.Start()
+		defer s.Stop()
+
+		appRealm, err := clients.Realm.CreateApp(
+			groupID,
+			cmd.inputs.Name,
+			createAppMetadata,
+		)
+		if err != nil {
+			return realm.App{}, err
+		}
+		return appRealm, nil
+	}
+	appRealm, err := createTemplateApp()
+	if err != nil {
+		return err
+	}
+
+	_, err = writeTemplateAppToLocal(clients.Realm, appRealm.ID, appRealm.GroupID, cmd.inputs.Template, rootDir)
+	if err != nil {
+		return err
+	}
+
+	output := newAppOutputs{
+		AppID:    appRealm.ClientAppID,
+		Filepath: rootDir,
+		Backend:  filepath.Join(rootDir, local.BackendPath),
+		URL:      fmt.Sprintf("%s/groups/%s/apps/%s/dashboard", profile.RealmBaseURL(), appRealm.GroupID, appRealm.ID),
+	}
+
+	frontendFolder := filepath.Join(rootDir, local.FrontendPath)
+	_, err = os.Stat(frontendFolder)
+	if err != nil && os.IsNotExist(err) {
+		output.Frontends = frontendFolder
+	}
+
+	for _, dsCluster := range dsClusters {
+		output.Clusters = append(output.Clusters, dataSourceOutputs{dsCluster.Name})
```

Comment:
instead of ranging over `dsClusters`, let's just append `dataSourceOutputs{output.DataSource.Name}` for now (since we've asserted there's only one cluster already)